src/Main.js

```diff
@@ -18,8 +18,12 @@ const Main = observer(() => {
       appStore.setAuth(true);
     }
     return () => {
-      ethereum.on('accountsChanged', () => null);
-      ethereum.on('chainChanged', () => null);
+      if (ethereum.removeListener) {
+        ethereum.removeListener('chainChanged', () => window.location.reload());
```

Comment:
This is wrong! You should remove previously defined function, not the new lambda.

Reply:
example here => https://github.com/NoahZinsmeister/web3-react/blob/v6/example/hooks.ts